### ISSUE TITLE
No text title on credit card payment method tab

### ISSUE THREAD

**Issue by haszari** [woocommerce/woocommerce-gutenberg-products-block#2438]
![Screen Shot 2020-05-07 at 2.20.12 PM.png](https://images.zenhubusercontent.com/5ddb3cb39529f10001e618e5/76b3e572-32c6-4274-828b-cfc9e7926b07)

Ideally the payment method tab should also include a text title, e.g. `Credit Card` alongside some card icons.

Here's a mockup from @garymurray - tagging in @LevinMedia for next steps:

![Screenshot 2020-05-07 at 13.59.45.png](https://images.zenhubusercontent.com/5ddb3cb39529f10001e618e5/e58b85ab-e6b4-4034-af66-81dfffb8afbc)

Summary of considerations / ideas:

- Include text title and icons on the tab - icons help make it recognisable as payment option.
- Currently lots of card icons, do we need them all on the tab?
   - Could use "generic" card icon so we don't need to have an icon for everything, e.g. show `generic visa mastercard` (focus on popular brands in tab).
   - In future may want to allow merchant to control which card icons are displayed (e.g. different brands are dominant in different regions/industries).
- Needs to work well on mobile.
- Need to explore what happens when there are many gateways - e.g. 3 may be common. `And more…` tab?
- Do we allow for tabs to wrap / have multiple lines?

And a related consideration: "Branding" the tab.

- The PayPal tab has a logo image in the mockup, should CC tab show that it is powered by Stripe?
- Maybe need some guidance around logos on tabs, so all gateways appear of similar prominence. A big bright logo might confuse users / make it harder to select a gateway.

Some specific suggestions/ideas from slack thread (ref below). 

> So we could potentially try include the title and icons on the tab?

> We could limit the number of displayed card icons, and just say “credit” rather the credit/debit to save on space

> I think it could wrap - but also aware it could look funky alongside PayPal which won’t wrap

> some merchants wanting control over what icons display

p1588823147160600-slack-woo-merchandising

**Comment by haszari** [comment]
Related quirk in cart block - card icons wrap on smaller window:

<img width="500" alt="Screen Shot 2020-05-08 at 12 27 45 PM" src="https://user-images.githubusercontent.com/4167300/81357317-71f73580-9127-11ea-8a90-bf1975e46f14.png">

Maybe we can progressively hide / refine the number of card icons displayed. 

Long term some merchants might want to tweak what's shown in cart, e.g. display gateway/mechanism logo or generic card logo, etc.

**Comment by LevinMedia** [comment]
@haszari Do we need the credit card icons in the cart? Doesn't seem necessary to me. Unless there's a good reason to have them there, I think we can remove. 

RE: Payment method tab

I'd be in favor of removing the icons from the tab, so it simply says "Credit card" we could add a credit card icon next to the label. 

Rather than in the tab, I'd run them horizontally under the credit card input form... it's an affordance for what you can enter into the field. This way they'd have much more space. Theoretically they should all be able to fit there, but as the viewport size decreases, we may need to add styles to shrink their size to keep all of them on one line.

**Comment by haszari** [comment]
For me the most important thing about card icons is they keep the shopper reassured that they will be able to pay with a card online. Secondary, that their specific type of card is supported.

> Do we need the credit card icons in the cart? Doesn't seem necessary to me. Unless there's a good reason to have them there, I think we can remove.

I'm not super qualified to comment, but I agree, I don't think card icons are essential in the cart. 

Having said that, I don't see a huge problem with them as they are, just could work better as the viewport gets smaller :)

Also, I'm not sure how this works – e.g. do we allow other payment methods to add icons/info to the cart (e.g. a PayPal icon) ? Maybe we can tweak the cart side of this in future, not urgent for v1. Seems like something that merchants might want to tweak. 

Let's focus this issue on the checkout payment step, which we need to get tidied up before v1.

> I'd be in favor of removing the icons from the tab, so it simply says "Credit card" we could add a credit card icon next to the label.
> Rather than in the tab, I'd run them horizontally under the credit card input form... it's an affordance for what you can enter into the field

I agree there needs to be some kind of card icon on the tab to provide that visual clue/reassurance to shopper. @garymurray mockup above limits this to top 3 icons, @LevinMedia has suggested using a generic card icon – either could work from my point of view.

This issue right now is blocked on design IMO – @LevinMedia can you review the discussion here and in [slack](p1588823147160600-slack-woo-merchandising) and come up with a final design for us to implement. cc @garymurray - can you review/feed back on the design too.

**Comment by LevinMedia** [comment]
@haszari lets remove the credit card icons from the cart page. 

Here are updates to the layout for checkout: 

Desktop:

<img width="691" alt="Screen Shot 2020-05-11 at 10 32 42 AM" src="https://user-images.githubusercontent.com/4500952/81592397-c3dfcb80-9372-11ea-9041-06cd3636b77b.png">

Mobile:

<img width="533" alt="Screen Shot 2020-05-11 at 10 33 05 AM" src="https://user-images.githubusercontent.com/4500952/81592436-d35f1480-9372-11ea-8b99-3e4af6637acb.png">


Updates as follows: 

- Remove credit card icons from the tab, and use the generic gridicon credit card next to Credit / Debit Card test. On mobile we loose the icon for space consideration.
- Card number, expiration, and cvc should be in their own containers. As you complete the entry in one input, the focus should automatically jump to the next. 
- Credit card icons are placed below the input fields. On desktop, the icons are 24px tall by proportional width with 8px between them. On mobile they're 16px tall by proportional width, with 8px between them. This will accommodate up to 10 credit card icons. 

Unresolved: 

@garymurray I did my best to look through the existing designs for this, but I didn't find anything that considered if there were more than two payment methods available. If that doesn't exist I think we'll have to do some additional exploration to cover that use case.

**Comment by garymurray** [comment]
> @garymurray I did my best to look through the existing designs for this, but I didn't find anything that considered if there were more than two payment methods available. If that doesn't exist I think we'll have to do some additional exploration to cover that use case. I don't think the tabs are going to scale well past two methods.

The idea was to change to showing them stacked vertically selectable as 'radio' options when we exceeded two methods.

> I also noticed the return to cart link was not included on the original mobile designs, so I added it back. 

👍

**Comment by garymurray** [comment]
> let's remove the credit card icons from the cart page.

What's the motivation/reasoning to remove the icons from the cart page? The reason they were included there was to give customer a clear idea that their payment method was supported during checkout.

**Comment by nerrad** [comment]
Per the conversation in slack (p1589281016321700-slack-C8X6Q7XQU), the decision is to work on the proposed changes for the icons in the checkout.

The cart icons (if further discussion is needed) can be decided in here, and a separate issue created if the decision is to remove them.

**Comment by mikejolley** [comment]
> Card number, expiration, and cvc should be in their own containers. As you complete the entry in one input, the focus should automatically jump to the next.

This isn't within scope, and is also gateway controlled. Stripe has options for both input types.
